perception/multi_object_tracker/include/multi_object_tracker/utils/utils.hpp

```diff
@@ -78,6 +80,129 @@ inline bool isLargeVehicleLabel(const uint8_t label)
   return label == Label::BUS || label == Label::TRUCK || label == Label::TRAILER;
 }
 
+/**
+ * @brief Get the Nearest Corner or Surface from detected object
+ *
+ * @param object
+ * @return int index
+ */
+inline int getNearestCornerSurface(
+  const double x, const double y, const double yaw, const double width, const double length,
+  const geometry_msgs::msg::Transform & self_transform)
+{
+  double x0, y0, xl, yl;
+  // get local vehicle pose
+  x0 = self_transform.translation.x;
+  y0 = self_transform.translation.y;
+
+  // localize self vehicle pose to object coordinate
+  // R.T (X0-X)
+  xl = std::cos(yaw) * (x0 - x) + std::sin(yaw) * (y0 - y);
+  yl = -std::sin(yaw) * (x0 - x) + std::cos(yaw) * (y0 - y);
+
+  // grid search
+  int xgrid, ygrid;
+  const int labels[3][3] = {{7, 0, 4}, {3, -1, 1}, {6, 2, 5}};
+  if (xl > length / 2.0) {
+    xgrid = 0;
+  } else if (xl > -length / 2.0) {
+    xgrid = 1;
+  } else {
+    xgrid = 2;
+  }
+  if (yl > width / 2.0) {
+    ygrid = 2;
+  } else if (yl > -width / 2.0) {
+    ygrid = 1;
+  } else {
+    ygrid = 0;
+  }
+
+  return labels[xgrid][ygrid];  // 0 to 7 + 1(null) value
+}
+
+/**
+ * @brief Get the Nearest Corner or Surface from detected object
+ *
+ * @param object
+ * @return int index
+ */
+inline int getNearestCornerSurfaceFromObject(
+  const autoware_auto_perception_msgs::msg::DetectedObject & object,
+  const geometry_msgs::msg::Transform & self_transform)
+{
+  // only work for BBOX shape
+  if (object.shape.type != autoware_auto_perception_msgs::msg::Shape::BOUNDING_BOX) {
+    return false;
```

Comment:
is it correct because return type is int?

Reply:
This is fixed in latest commit https://github.com/autowarefoundation/autoware.universe/pull/2238/commits/0cf095d9ac83f0afd203c57c7cf366607fc7a46d.